front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -163,6 +185,48 @@ export function ProviderManagementModal({
           })}
         </ContextItem.List>
       </div>
+      <div className="flex flex-row items-center gap-4 px-4 pt-4">
+        <div className="s-text-sm font-semibold">Embedding Provider:</div>
+        <DropdownMenu>
+          <DropdownMenu.Button>
+            <Tooltip label="Please contact us if you are willing to change this setting.">
+              <Button
+                type="select"
+                labelVisible={true}
+                label={
+                  embeddingProvider
+                    ? prettyfiedProviderNames[embeddingProvider]
+                    : prettyfiedProviderNames["openai"]
+                }
+                variant="secondary"
+                hasMagnifying={false}
+                size="sm"
+                disabled={true}
+              />
+            </Tooltip>
+          </DropdownMenu.Button>
+          <DropdownMenu.Items origin="topRight">
+            {EMBEDDING_PROVIDER_IDS.map((provider) => (
+              <DropdownMenu.Item
+                key={provider}
+                label={prettyfiedProviderNames[provider]}
+                onClick={() => {
+                  setDefaultEmbeddingProvider(provider);
+                }}
+              />
+            ))}
+          </DropdownMenu.Items>
+        </DropdownMenu>
+      </div>
+      <div className="px-4 pt-2 text-sm text-gray-500">
+        Embedding turns knowledge into a code, making it easy to search and find
```

Comment:
Embedding models are used to create numerical representations of your data powering the semantic search capabilities of your assistants.